File: Day1/R/Solution.R

#Loops: https://www.datacamp.com/tutorial/tutorial-on-loops-in-r
#Loops: https://intro2r.com/loops.html
#Print: https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/sprintf
#Print: https://www.geeksforgeeks.org/printing-output-of-an-r-program/ 
#Vector compare: https://statisticsglobe.com/compare-vectors-and-find-differences-in-r

getsol <- function(reqsum, numberlist) {
  
  totalpass <- 0
  
  for (num1 in numberlist){
    totalpass <- totalpass + 1      
    tmp = (reqsum - num1)
    if(tmp %in% numberlist){
      print(paste("Combination: ", num1, tmp))
    }
  }
  print(paste("Total number of passes: ", totalpass))
}

getsol(5,c(1,2,3,4,5))

File: Day1/Python/Solution.py
#Function Arguments: https://www.toppr.com/guides/computer-science/programming-with-python/lists/lists-as-arguments/ 
#Print formatting: https://pyformat.info/


def getsol(reqsum, *numberlist):
    print("Need a 2 number combination that summs upto {0} from the list {1}".format(reqsum,numberlist))

    totalpass = 0
    for num1 in numberlist:
        tmp = reqsum - num1
        if(tmp in numberlist):
            totalpass = totalpass + 1
            print("{0} + {1} = {2}".format(num1, tmp, reqsum))
    print("Total passes: {0}".format(totalpass))


getsol(5,*[1,2,3,4,5])